File: okj/szoftverfejleszto/nobel/java/Dij.java
public class Dij{
    
    public final int evszam;
    public final String tipus;
    public final String keresztNev;
    public final String vezetekNev;
    
    public Dij(String sor) {
        var split = sor.split(";");
        
        evszam = Integer.parseInt(split[0]);
        tipus = split[1];
        keresztNev = split[2];
        vezetekNev = split.length == 4 ? split[3] : "";
    }
}

File: okj/szoftverfejleszto/nobel/c#/Dij.cs
public class Dij {

    public readonly int ev;
    public readonly string tipus;
    public readonly string keresztNev;
    public readonly string vezetekNev;
    
    public Dij(string sor) {
        var split = sor.Split(';');
        
        ev = int.Parse(split[0]);
        tipus = split[1];
        keresztNev = split[2];
        vezetekNev = split.Length == 4 ? split[3] : "";
    }
}